configure

```diff
@@ -5351,6 +5351,11 @@ EOF
 
 if $CPP $CPPFLAGS conftest.c >conftest.out 2>/dev/null; then
   PLATFORM_TRIPLET=`grep -v '^#' conftest.out | grep -v '^ *$' | tr -d ' 	'`
+  case "$build_os" in
```

Comment:
If I'm not mistaken, `PLATFORM_TRIPLET` should refer to the target platform on which cpython will run. If that is the case, the musl libc vs. glibc decision should be based on `$host_os` rather than `$build_os` as the former is based on autoconf's `AC_CANONICAL_HOST` macro[^1] which refers to the target platform that might differ from build platform in case of cross-compilation.

For example, the decision based on `$build_os` leads to 
cpython 3.9.13 build failing[^2] when cross-compiling on AMD64 Linux with glibc for mpc8548 Linux (OpenWrt) with musl. (build runs correctly if `$host_os` is used)

[^1]: https://www.gnu.org/software/autoconf/manual/autoconf-2.68/html_node/Canonicalizing.html
[^2]: "internal configure error for the platform triplet, please file a bug report"

Reply:
Yes, the code is wrong. In fact the entire block should be revisited. We really should check for ``__GLIBC__`` and use a heuristic to detect musl. Could you please open a new issue for the problem?

Reply:
No problem, but there's https://github.com/python/cpython/issues/87278 already. Is yet another issue needed?

Reply:
Created https://github.com/python/cpython/issues/95855